### PR TITLE
Fixed Julia 0.6 deprecations (and removed Julia 0.4 support)

### PR DESCRIPTION
**Pull request by mauro3**

### CODE REVIEW

**Comment by c42f** [comment]
Looks great to me, thanks.

**Comment by mauro3** [comment]
Thanks for the swift work!

**Comment by c42f** [comment]
No worries! This time I've remembered to tag a new version:
https://github.com/JuliaLang/METADATA.jl/pull/10102

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-julia 0.4
+julia 0.5
 BinDeps
-Compat 0.8
+Compat 0.26.0
```
Comment (tkelman) [inline REQUIRE]: only need 0.17 for abstract type

Comment (mauro3) [inline REQUIRE]: Fixed in 2c64559d1f266b7f12e4c0531a70994279d9d2d3.  I'll update the METADATA PR in a sec.

**Pull request merged**